swig/python/CMakeLists.txt

```diff
@@ -218,6 +230,18 @@ add_custom_command(TARGET ${PYTHON_MODULE_NAME} POST_BUILD     # Adds a post-bui
         "${FLAMEGPU_ROOT}/include"                                     # <--this is in-dir
         "${PYTHON_LIB_OUTPUT_DIRECTORY}/${PYTHON_MODULE_NAME}/include" # <--this is out-dir path
     ) 
+
+# Create code generator directory
+add_custom_command(TARGET ${PYTHON_MODULE_NAME} POST_BUILD 
+    COMMAND ${CMAKE_COMMAND} -E make_directory "${PYTHON_LIB_OUTPUT_DIRECTORY}/${PYTHON_MODULE_NAME}/codegen"
+)
+# Copy python codegen source files to library output directory ready for packaging
+foreach( file_py ${PYTHON_CODEGEN_SRC_FILES})
+  add_custom_command(TARGET ${PYTHON_MODULE_NAME} POST_BUILD
```

Comment:
This does not behave correctly for incremental builds where only parts of `codegen` are modified.

Under linux (and presumably windows too), (probably) because this is a `POST_BUILD` event on the `pyflamegpu` target (`PYTHON_MODULE_NAME`) the files are only copied when the pyflamegpu target has been built. 

I.e. if the pyflamegpu module is up to date so it does not need to be built, modifications to the codegen are not copied across. 

1. Build `pyflamegpu`
2. Edit `swig/pythong/codegen/codegen.py`
3. Invoke a build (not a clean and build / rebuild in VS terms)
4. Check the codegen file in the venv (`lib/<build_type>/python/venv/lib, it will not have been updated. (i.e. `lib/Release/python/pyflamegpu/codegen/codegen.py`)


Instead, iff a clean build is done, or a change which would invoke an incremental build of pyflamegpu is triggered the file is updated (and so is the version in the venv)

This seems to suggest that the `DEPENDS` statement below is insufficient, though i'm not sure what the correct way to fix this is off the top of my head. 

It could also require an `OUTPUTS` section adding to this custom command based on a quick read of the [docs](https://cmake.org/cmake/help/v3.0/command/add_custom_command.html), but I have a feeling its more to do with the POST_BUILD not being triggered (Editing `pyflamegpu.i` and triggering a build does work). 

This might be a fairly involved restrucutre to get this to trigger without a pyflamegpu.i rebuild (may need an intermediate target to avoid the long compile step, but this is speculation).

Reply:
see https://github.com/FLAMEGPU/FLAMEGPU2/pull/887

Reply:
Resolved by PR #887 now merged int this branch